app/controllers/courses/subjects_controller.rb

```diff
@@ -91,11 +91,39 @@ def build_course
     end
 
     def build_course_params
+      selected_master = params[:course][:master_subject_id]
+      selected_subordinate = nil
+      selected_subordinate = params[:course][:subordinate_subject_id] if params[:course][:subordinate_subject_id].present?
+      previous_subject_selections = params[:course][:subjects_ids]
+
       params[:course][:subjects_ids] = []
-      params[:course][:subjects_ids] << params[:course][:master_subject_id]
-      params[:course][:subjects_ids] << params[:course][:subordinate_subject_id] if params[:course][:subordinate_subject_id].present?
+      params[:course][:subjects_ids] << selected_master
+      params[:course][:subjects_ids] << selected_subordinate if selected_subordinate
       params[:course].delete(:master_subject_id)
       params[:course].delete(:subordinate_subject_id)
+
+      build_new_course # to get languages edit_options
+
+      if modern_language_selected?
```

Comment:
There seem to be two methods with the same name but this one is being declared inside another method, was this intentional?

Reply:
`if` != `def` :laughing: